mantle/cmd/kola/testiso.go

```diff
@@ -0,0 +1,325 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/coreos/mantle/util"
+	"github.com/pkg/errors"
+
+	"github.com/coreos/mantle/sdk"
+
+	ignconverter "github.com/coreos/ign-converter"
+	ignv3types "github.com/coreos/ignition/v2/config/v3_0/types"
+	"github.com/spf13/cobra"
+
+	"github.com/coreos/mantle/kola"
+	"github.com/coreos/mantle/platform"
+)
+
+var (
+	cmdTestIso = &cobra.Command{
+		Run:    runTestIso,
+		PreRun: preRun,
+		Use:    "testiso",
+		Short:  "Test a CoreOS PXE boot or ISO install path",
+	}
+	// TODO expose this as an API that can be used by cosa too
+	consoleKernelArgument = map[string]string{
+		"amd64-usr":   "ttyS0",
+		"ppc64le-usr": "hvc0",
+		"arm64-usr":   "ttyAMA0",
+		"s390x-usr":   "ttysclp0",
+	}
+)
+
+func init() {
+	root.AddCommand(cmdTestIso)
+}
+
+func runTestIso(cmd *cobra.Command, args []string) {
+	if err := doTestIso(cmd, args); err != nil {
+		fmt.Fprintf(os.Stderr, "%s\n", err)
+		os.Exit(1)
+	}
+}
+
+func doTestIso(cmd *cobra.Command, args []string) error {
+	if kola.CosaBuild == nil {
+		return fmt.Errorf("Must provide --cosa-build")
+	}
+
+	if kola.CosaBuild.BuildArtifacts.Kernel.Path != "" {
+		return errors.Wrapf(testLegacyInstaller(), "testing legacy installer")
```

Comment:
Hmm, can we unpack this into the more traditional flow for checking for errors? Nesting the call here and relying on the `nil` no-op handling feels funny.

Reply:
This [is documented](https://godoc.org/github.com/pkg/errors#Wrap) 
> If err is nil, Wrap returns nil. 

Presumably for this reason, but eh, I see your point, so changed.